src/pool/srv_target.c

```diff
@@ -397,6 +403,7 @@ pool_child_delete_one(void *uuid)
 
 	D_ASSERT(d_list_empty(&child->spc_cont_list));
 	d_list_del_init(&child->spc_list);
+	ds_stop_chkpt_ult(child);
```

Comment:
I think we'd better stop checkpoint ULT when ds_pool_child drop the last reference in ds_pool_child_put(), otherwise, there could be potential deadlock issue:

1. I/O request ULT calls bio_wal_reserve(), it find out that WAL space is insufficient, then bio_wal_reserve() yield and wait for checkpoint to reclaim WAL space.
2. Pool is destroyed by force, pool_child_delete_one() is called to stop checkpoint ULT.
3. bio_wal_reserve() will wait forever since the WAL space won't be reclaimed.

I'm not sure if we still support force pool destroy today, but seems to me stopping such internal ULTs at the end (after all other users dropped reference) is a good practice.